### ISSUE TITLE
OpenMSX crashes after deleting imgui.ini and restarting

### ISSUE THREAD

**Issue by pvmm** [openMSX/openMSX#1598]
### How to reproduce:

1. delete share/imgui.ini
2. start OpenMSX
3. open ImGui memory widget

### OpenMSX crashes with the following message:
/usr/bin/../lib/gcc/x86_64-redhat-linux/13/../../../../include/c++/13/debug/safe_iterator.h:328:
In function:
    _Safe_iterator<_Iterator, _Sequence, _Category> &
    gnu_debug::_Safe_iterator<gnu_cxx::
    normal_iterator<openmsx::ImGuiPartInterface **, std::
    vector<openmsx::ImGuiPartInterface *>>, 
    std::vector<openmsx::ImGuiPartInterface *>, 
    std::forward_iterator_tag>::operator++() [_Iterator = gnu_cxx::
    normal_iterator<openmsx::ImGuiPartInterface **, std::
    vector<openmsx::ImGuiPartInterface *>>, _Sequence = 
    std::vector<openmsx::ImGuiPartInterface *>, _Category = 
    std::forward_iterator_tag]

Error: attempt to increment a singular iterator.

Objects involved in the operation:
    iterator "this" @ 0x7ffc2dcdf688 {
      type = gnu_cxx::normal_iterator<openmsx::ImGuiPartInterface**, std::vector<openmsx::ImGuiPartInterface*, std::allocator<openmsx::ImGuiPartInterface*> > > (mutable iterator);
      state = singular;
      references sequence with type 'std::debug::vector<openmsx::ImGuiPartInterface*, std::allocator<openmsx::ImGuiPartInterface*> >' @ 0x23ca6c0
    }
./start.sh: line 4: 80858 Aborted                 (core dumped) OPENMSX_HOME=/home/pedro/Projects/msx/openMSX ./derived/openmsx -machine msx2

### And here is the backtrace:

[sigabort.txt](https://github.com/openMSX/openMSX/files/14744446/sigabort.txt)

**Comment by pvmm** [comment]
Seems like parts vector is changing during a loop. If I create a temporary parts vector and loop through it, the crash doesn't happen:
 
```
void ImGuiManager::paintImGui()
{
        std::vector<ImGuiPartInterface*> partsCopy = parts;
        (...)
        auto drawMenu = [&]{
                for (auto* part : partsCopy) { 
                        part->showMenu(motherBoard);
                }
        };
```

**Comment by pvmm** [comment]
Closed because I can't reproduce it anymore. Fixed by 93e82a566b536e2b6bcb4da510ebf02132a24ecc
